src/Microsoft.OData.Edm/Csdl/Parsing/Common/XmlDocumentParser.cs

```diff
@@ -493,7 +493,11 @@ private void ReportUnexpectedNodeType(XmlNodeType nodeType)
 
         private void ReportUnexpectedElement(CsdlLocation errorLocation, string elementName)

```

Comment:
>ReportUnexpectedElement [](start = 21, length = 23)

Could the if go in the to places where this is called instead? All other Report functions always report, this one only does sometimes. IMO, it makes the code calling this function harder to read/understand.